frontend_tests/node_tests/compose.js

```diff
@@ -46,12 +48,69 @@ people.initialize_current_user(me.user_id);
 people.add(alice);
 people.add(bob);
 
-(function test_clear_all_everyone_warnings() {
-    $("#compose-all-everyone").show();
-    $("#send-status").show();
-    compose.clear_all_everyone_warnings();
-    assert(!$("#compose-all-everyone").visible());
-    assert(!$("#send-status").visible());
+i18n.init({
+    nsSeparator: false,
+    keySeparator: false,
+    interpolation: {
+        prefix: "__",
+        suffix: "__",
+    },
+    lng: 'en',
+});
+
+(function test_validate() {
+    $("#compose-send-button").removeAttr('disabled');
+    $("#compose-send-button").focus();
+    $("#sending-indicator").hide();
+    assert(!compose.validate());
+    assert(!$("#sending-indicator").visible());
+    assert(!$("#compose-send-button").is_focused());
+    assert.equal($("#compose-send-button").attr('disabled'), undefined);
+    assert.equal($('#error-msg').html(), 'You have nothing to send!');
+
+    $("#new_message_content").val('foobarfoobar');
+    var zephyr_visible = false;
+    $("#zephyr-mirror-error").is = function () {
+        if (!zephyr_visible) {
+            zephyr_visible = true;
+            return true;
+        }
+        return false;
+    };
+    assert(!compose.validate());
+    assert(zephyr_visible);
+    assert.equal($('#error-msg').html(), 'You need to be running Zephyr mirroring in order to send messages!');
+
+    compose_state.set_message_type('private');
+    compose_state.recipient('');
+    assert(!compose.validate());
+    assert.equal($('#error-msg').html(), 'Please specify at least one recipient');
+
+    compose_state.recipient('foo@zulip.com');
+    global.page_params.realm_is_zephyr_mirror_realm = true;
+    assert(compose.validate());
+
+    global.page_params.realm_is_zephyr_mirror_realm = false;
+    assert(!compose.validate());
+    assert.equal($('#error-msg').html(), 'The recipient foo@zulip.com is not valid');
+
+    compose_state.recipient('foo@zulip.com,alice@zulip.com');
+    assert(!compose.validate());
+    assert.equal($('#error-msg').html(), 'The recipients foo@zulip.com,alice@zulip.com are not valid');
+
+    compose_state.recipient(',');
```

Comment:
nopes, I guess I could have done something better here like
`alice@zulip.com,` basically comma is needed to generate an empty string so that we can reach some line of code. I don't remember exactly which line it was, but after taking a look at code I can point out.